cg/store/api/find_business_data.py

```diff
@@ -713,3 +713,24 @@ def get_running_cases_in_pipeline(self, pipeline: Pipeline) -> List[Family]:
             cases=self._get_query(table=Family),
             pipeline=pipeline,
         ).all()
+
+    def get_cases_not_analysed_by_sample_internal_id(
+        self,
+        sample_internal_id: str,
+    ) -> List[Family]:
+        """Get cases to analyse by sample internal id."""
+
+        query = self._get_join_case_and_sample_query()
+
+        not_analysed_cases = apply_case_filter(
+            cases=query,
+            filter_functions=[
+                CaseFilter.GET_NOT_ANALYSED,
+            ],
+        )
+
+        return apply_sample_filter(
+            samples=not_analysed_cases,
+            filter_functions=[SampleFilter.FILTER_BY_INTERNAL_ID],
+            internal_id=sample_internal_id,
+        ).all()
```

Comment:
Is this filtering by the internal id of the sample or the case? I think the CaseFilter function should be used instead of the Sample to avoid confusion. If the intended behaviour is to filter by sample I think this is not doing it. I think is taking the internal id of the cases. Better double check :)

Reply:
The filtering should be done with the sample internal id and the function is called with sample.internal_id, so it should be ok!